src/components/InfiniteSlidePanel/InfiniteSlidePanel.jsx

```diff
@@ -0,0 +1,121 @@
+import _ from 'lodash';
+import React from 'react';
+import { lucidClassNames } from '../../util/style-helpers';
+import { createClass, getFirst, omitProps } from '../../util/component-types';
+import SlidePanel from '../SlidePanel/SlidePanel';
+
+const cx = lucidClassNames.bind('&-InfiniteSlidePanel');
+
+const {
+	func,
+	node,
+	number,
+	oneOfType,
+	string,
+} = React.PropTypes;
+
+const modulo = (n, a) => (a - (n * Math.floor(a/n)));
```

Comment:
Maybe pull this function into a new `util` since it's used in both `SlidePanel`s.